DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/SRSnapshotTests.swift

```diff
@@ -292,9 +332,12 @@ final class SRSnapshotTests: SnapshotTestCase {
         ]
 
         for fixture in navBarFixtures {
-            show(fixture: fixture)
+            let navController = show(fixture: fixture) as! TestNavigationController
+            navController.pushNextView()
+            // Wait for view controller to be pushed onto the navigation stack.
+            wait(seconds: 0.5)
```

Comment:
question/ This might become flaky. Is there a way to perform the segue without animation, so the `pushNextView()` could be blocking API that guarantees expected state after return?

Reply:
I tried to push the view without animation, but it's still necessary to wait for at least 0.1 seconds.

Here's an alternative solution (I haven't tried it): We can create a `UIViewController` subclass with a handler that notifies the navigation controller once `viewDidAppear` has been called. Using `XCTWaiter`, we can then wait for the view to appear. This approach ensures we wait for the view to actually appear rather than relying on an arbitrary delay. It also avoids unnecessarily long waits, making the tests run a bit faster.

While we could consider removing this navigation test altogether, doing so would reduce the test coverage of the recorder.

A small note: although this issue is worth addressing, there are many instances of `wait(seconds:)` in these snapshot tests.

Reply:
I wonder if [`UINavigationController.setViewControllers(_:animated:)`](https://developer.apple.com/documentation/uikit/uinavigationcontroller/1621861-setviewcontrollers) could help (AFAIK it is used by push / performSegue under the hood).

> there are many instances of wait(seconds:) in these snapshot tests

Fair point 👌 and the main reason I don't consider this a blocking feedback. I'm totally ok with what is implemented here 👍.

Reply:
Unfortunately it's not working; a delay is still necessary. I was able to reduce the delay for the tint color test to 0.1 seconds. I ran both tests repeatedly, and they passed ✅